ote/src/cljs/ote/app/controller/front_page.cljs

```diff
@@ -78,12 +80,34 @@
 
   OpenUserMenu
   (process-event [_ app]
-    (assoc-in app [:ote-service-flags :user-menu-open] true) app)
+    (-> app
+      (assoc-in [:ote-service-flags :user-menu-open]
+                (if (get-in app [:ote-service-flags :user-menu-open]) false true))
+      (assoc-in [:ote-service-flags :header-open] false)
+      (assoc-in [:ote-service-flags :lang-menu-open] false)))
 
   OpenHeader
   (process-event [_ app]
-    (assoc-in app [:ote-service-flags :header-open]
-              (if (get-in app [:ote-service-flags :header-open]) false true)))
+    (-> app
+      (assoc-in [:ote-service-flags :header-open]
+              (if (get-in app [:ote-service-flags :header-open]) false true))
+      (assoc-in [:ote-service-flags :user-menu-open] false)
+      (assoc-in [:ote-service-flags :lang-menu-open] false)))
+
+  OpenLangMenu
+  (process-event [_ app]
+    (-> app
+      (assoc-in [:ote-service-flags :lang-menu-open]
```

Comment:
`(update-in [:ote-service-flags :lang-menu-open] not)` is a simpler way to negate a boolean flag value.